### PR TITLE
Remove old `_optimize` method and the test

### PR DESCRIPTION
**Pull request by YigitElma**

### CODE REVIEW

**Review by f0uriest — APPROVED** [review]

**Comment by ddudt** [comment]
It's fine for this PR because I would have approved it, but remember that in general our unofficial policy is to get 2 approvals before merging. More eyes help catch things that one person misses.

**Comment by YigitElma** [comment]
> It's fine for this PR because I would have approved it, but remember that in general our unofficial policy is to get 2 approvals before merging. More eyes help catch things that one person misses. 

Oh sorry, I thought one approval is fine in general. Also we have already talked about this during meeting, that's why I just merged it in.

**Pull request merged**
